frontend/src/components/cart-page/CartSummarySection.tsx

```diff
@@ -0,0 +1,72 @@
+import { useContext, useMemo } from 'react'
+import { useNavigate } from 'react-router-dom';
+
+import { ShoppingCartContext } from '../../contexts/shopping-cart-context/shoppingCartContext';
+import { Box, Button, Card, CardContent, Divider, Typography } from "@mui/material";
+
+import cartStyles from './cartStyles';
+const {
+    totalCardStyle,
+    buttonContainerStyle,
+    buttonStyle,
+} = cartStyles;
+
+const CartSummarySection = () => {
+    const navigate = useNavigate();
+    const { cartItems, userSignin } = useContext(ShoppingCartContext);
+
+    const SHIPPING_PATH = "/shipping";
+    const SIGNIN_PATH = "/user/signin";
+
+    const handleProceedToCheckout = () => {
+        userSignin ? navigate(SHIPPING_PATH) : navigate(`${SIGNIN_PATH}?redirect=${SHIPPING_PATH}`);
+    };
+
+
+    const calculateTotalPrice = useMemo(() => {
+        return cartItems.reduce((a, c) => a + c.price * c.quantity, 0);
+    }, [cartItems]);
```

Comment:
isn't this similar to this [util fn](https://github.com/Ameer-Alaswad/E-commerce/blob/adcd66582f11f2887e3504f1fd4ce707a820423d/frontend/src/components/checkout/utils.tsx#L44)?
maybe we can import it instead of duplicating the code 💭

Reply:
I totally forgot about this utility function, thx for the reminder.